### PR TITLE
 Add Nakadi-Docker based integration tests

### LINKED ISSUE #271: Add Nakadi-Docker based integration tests
Add testcontainers-based Nakadi integration tests

### PR DESCRIPTION
**Pull request by otrosien**
Adds docker-compose based integration tests for the three httpclient integrations.

Closes #271 , builds ontop of #276 (to be rebased once merged)

Because of cyclic dependencies (`fahrschein -> fahrschein-http-simple -> fahrschein-test-support -> fahrschein`), I cannot put the `AbstractRequestFactoryTest` into `fahrschein-test-support`, but we could put the e2e-suite into a new module `fahrschein-e2e-test`, WDYT?

The metadata chages are also up to debate, I just realised there were some missing fields when running jackson with `FAIL_ON_UNKNOWN_PROPERTIES`.

TODO

* [x] Documentation (README)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -66,8 +66,8 @@
     </distributionManagement>
 
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
```
Comment (otrosien) [inline pom.xml:69]: oops.

**Review by othon2000 — CHANGES_REQUESTED** [review]
Thanks for this.

I'm more in favor of having a new `fahrschein-e2e-test` module, as the test contents are not really  related to `fahrschein-example`.

File: README.md [hunk 2]
```diff
@@ -271,6 +271,14 @@ For local development, Fahrschein requires:
 
 When developing, make sure to run unit and integration tests with `mvn verify`.
 
+The `fahrschein-e2e-test` module has end-to-end tests using `docker-compose`. If you wish to leave the Docker containers running between tests, you can also bring up docker-compose manually using its [docker-compose.yaml](fahrschein-e2e-test/src/test/resources/docker-compose.yaml), before running the tests.
```
Comment (othon2000) [inline README.md:274]:
```suggestion
The `fahrschein-e2e-test` module has end-to-end tests using `docker-compose`. If you wish to leave the Docker containers running between tests, you can also bring up docker-compose manually using its [docker-compose.yaml](fahrschein-e2e-test/src/test/resources/docker-compose.yaml), before running the tests. In either case, you will need the port `8080` to be available for Nakadi to run.
```

**Review by othon2000 — APPROVED** [review]

**Comment by othon2000** [comment]
:+1:

**Comment by otrosien** [comment]
👍

**Pull request merged**